src/assets/Generator.Shared/RequestUriBuilderExtensions.cs

```diff
@@ -6,120 +6,137 @@
         {
-            builder.AppendQuery(name, TypeFormatters.ToString(value, format), escape);
+            builder.AppendQuery(name, ConvertToString(value, format), escape);
         }
 
         public static void AppendQuery(this RequestUriBuilder builder, string name, double value, bool escape = true)
         {
-            builder.AppendQuery(name, value.ToString(TypeFormatters.DefaultNumberFormat, CultureInfo.InvariantCulture), escape);
+            builder.AppendQuery(name, ConvertToString(value), escape);
         }
 
         public static void AppendQuery(this RequestUriBuilder builder, string name, decimal value, bool escape = true)
         {
-            builder.AppendQuery(name, value.ToString(TypeFormatters.DefaultNumberFormat, CultureInfo.InvariantCulture), escape);
+            builder.AppendQuery(name, ConvertToString(value), escape);
         }
 
         public static void AppendQuery(this RequestUriBuilder builder, string name, int value, bool escape = true)
         {
-            builder.AppendQuery(name, value.ToString(TypeFormatters.DefaultNumberFormat, CultureInfo.InvariantCulture), escape);
+            builder.AppendQuery(name, ConvertToString(value), escape);
         }
 
         public static void AppendQuery(this RequestUriBuilder builder, string name, long value, bool escape = true)
         {
-            builder.AppendQuery(name, value.ToString(TypeFormatters.DefaultNumberFormat, CultureInfo.InvariantCulture), escape);
+            builder.AppendQuery(name, ConvertToString(value), escape);
         }
 
         public static void AppendQuery(this RequestUriBuilder builder, string name, TimeSpan value, bool escape = true)
         {
-            builder.AppendQuery(name, XmlConvert.ToString(value), escape);
+            builder.AppendQuery(name, ConvertToString(value), escape);
         }
 
         public static void AppendQuery(this RequestUriBuilder builder, string name, byte[] value, string format, bool escape = true)
         {
-            builder.AppendQuery(name, TypeFormatters.ToString(value, format), escape);
+            builder.AppendQuery(name, ConvertToString(value, format), escape);
         }
 
         public static void AppendQuery(this RequestUriBuilder builder, string name, Guid value, bool escape = true)
         {
-            builder.AppendQuery(name, value.ToString(), escape);
+            builder.AppendQuery(name, ConvertToString(value), escape);
         }
 
-        public static void AppendQueryDelimited<T>(this RequestUriBuilder builder, string name, IEnumerable<T> value, string delimiter, bool escape = true)
+        public static void AppendQueryDelimited<T>(this RequestUriBuilder builder, string name, IEnumerable<T> value, string delimiter, bool escape = true) where T : notnull
         {
-            builder.AppendQuery(name, string.Join(delimiter, value), escape);
+            var stringValues = value.Select(v => ConvertToString(v));
```

Comment:
Using Linq sometimes has fairly negative perf implications - have you looked at the difference in performance from using this approach vs. adding the overloads we discussed in the issue?  Do we know which approach is favorable from a performance perspective?

Reply:
we could introduce a string builder here instead

Reply:
Well I cannot find a way to do this without linq - 
we could do it in an array but it need us to know when we are getting the last since we need to omit an extra `,`
I do not think we could do that and check if the string ends with a comma because you will never know if the input values contain comma at the end.
when we are trying to identify if the element is the last, we will have to use linq method - either ToArray, ToList or Count.
Therefore I believe linq method here is inevitable.
Do you have a recommendation?

Reply:
Well since this discussion is here for quite a while, and if we have better solution for this linq issue, we could do that later.
Therefore I would like to merge this PR for now, and then we could introduce some other changes to polish